### PR TITLE
[MM-46502] Add API docs for top dms, top threads

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

Documents the  following GET API endpoints.
 - `api/v4/teams/:team_id:/top/threads` 
 - `api/v4/users/me/top/threads` 
 - `api/v4/users/me/top/dms` 
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-46502

#### Related To
https://github.com/mattermost/mattermost-server/pull/20195
https://github.com/mattermost/mattermost-server/pull/20690

### CODE REVIEW

File: v4/source/insights.yaml [hunk 1]
```diff
@@ -230,7 +230,7 @@
             application/json:
               schema:
                 type: object
-                $ref: "#/components/schemas/TopReactionList"
+                $ref: "#/components/schemas/TopChannelList"
```
Comment (plant99) [inline v4/source/insights.yaml:233]: Typo from last iteration.

**Comment by plant99** [comment]
Ben, Martin, have added you both as reviewers but only one from either of you would suffice. Carrie's one is necessary.

**Review by cwarnermm — APPROVED** [review]
Thanks, @plant99!

File: v4/source/definitions.yaml [hunk 2]
```diff
@@ -853,6 +853,91 @@ components:
...
+          type: integer
+          format: int64
+    TopThread:
+      type: object
+      properties:
+        post:
+          $ref: "#/components/schemas/Post"
+        channel_id:
+          type: string
+        display_name:
```
Comment (mkraft) [inline v4/source/definitions.yaml:879]:
```suggestion
        channel_display_name:
```

File: v4/source/definitions.yaml [hunk 3]
```diff
@@ -853,6 +853,91 @@ components:
...
+          type: string
+        name:
```
Comment (mkraft) [inline v4/source/definitions.yaml:881]:
```suggestion
        channel_name:
```

File: v4/source/definitions.yaml [hunk 4]
```diff
@@ -853,6 +853,91 @@ components:
...
+          type: string
+        last_name:
+          type: string
+        nick_name:
+          type: string
+        username:
+          type: string
+        create_at:
+          type: integer
+          format: int64
```
Comment (mkraft) [inline v4/source/definitions.yaml:916]: Could everything except `position` be referenced using the `InsightUserInformation` indirection ref?

File: v4/source/definitions.yaml [hunk 5]
```diff
@@ -853,6 +853,91 @@ components:
...
+        position:
+          type: string
+    TopDM:
+      type: object
+      properties:
+        message_count:
```
Comment (mkraft) [inline v4/source/definitions.yaml:922]:
```suggestion
        post_count:
```

File: v4/source/insights.yaml [hunk 6]
```diff
@@ -289,3 +289,175 @@
...
+      summary: Get a list of the top threads for a user.
+      description: |
+        Get a list of the top threads from public and private channels (the user is a member of and participating in the thread) for a given user.
+        ##### Permissions
+        Must be logged in as the user.
+      operationId: GetTopThreadsForUser
+      parameters:
+        - name: user_id
+          in: path
+          description: User GUID
```
Comment (mkraft) [inline v4/source/insights.yaml:359]: I think we currently only accept `me`, so this description could lead to confusion.

File: v4/source/insights.yaml [hunk 7]
```diff
@@ -289,3 +289,175 @@
...
+      summary: Get a list of the top dms for a user.
+      description: |
+        Get a list of the top dms for a given user.
+        ##### Permissions
+        Must be logged in as the user.
+      operationId: GetTopDMsForUser
+      parameters:
+        - name: user_id
+          in: path
+          description: User GUID
```
Comment (mkraft) [inline v4/source/insights.yaml:423]: I think we currently only accept `me`, so this description could lead to confusion.

**Review by mkraft — CHANGES_REQUESTED** [review]

File: v4/source/definitions.yaml (hunk 4 above)
Comment (plant99) [reply v4/source/definitions.yaml]: Thanks, fixed this.

**Pull request merged**